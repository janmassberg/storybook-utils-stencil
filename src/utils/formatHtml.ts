import { format } from "prettier/standalone";
import parserHtml from "prettier/parser-html";
import parserBabel from "prettier/parser-babel";

/**
 * Formats HTML source code.
 * @param {string} code The source code
 */
export const formatHtml = (code: string): string =>
    format(code, {
        parser: "html",
        plugins: [parserHtml, parserBabel],
        htmlWhitespaceSensitivity: "ignore",
    });
