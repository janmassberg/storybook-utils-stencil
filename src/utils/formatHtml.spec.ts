import { formatHtml } from "./formatHtml";

describe("formatHtml", () => {
    it("converts string to camel case correctly", () => {
        expect(
            formatHtml(
                `<div
class="my-container"
data-id="test-id"
data-name="test-name"><span>test content</span></div>`
            )
        ).toMatchSnapshot();
    });
});
