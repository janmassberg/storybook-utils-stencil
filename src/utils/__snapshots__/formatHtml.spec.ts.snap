// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatHtml converts string to camel case correctly 1`] = `
"<div class=\\"my-container\\" data-id=\\"test-id\\" data-name=\\"test-name\\">
  <span>test content</span>
</div>
"
`;
